File: android/app/src/main/java/io/khkhm/notebook/domain/NoteBook.kt
package io.khkhm.notebook.domain

import com.google.gson.annotations.SerializedName
import io.khkhm.notebook.data.BaseResponse
import java.io.Serializable
import java.util.*
import kotlin.collections.ArrayList


data class NoteBook(var name: String,
                    val date: Date = Date(),
                    var color: Color,
                    val notes: ArrayList<Note> = ArrayList(),
                    @SerializedName("_id") val id: String = "") : Serializable, BaseResponse() {
}

File: server/models/notebook.js
var mongoose = require('mongoose');

var Schema = mongoose.Schema;

var NoteSchema = new Schema({
    title: 'String',
    date: 'Date',
    text: 'String',
    color: 'String'
})

var NoteBookSchema = new Schema({
    name: 'String',
    date: 'Date',
    color: 'String',
    notes: [NoteSchema]
})

module.exports = {
    notebook: mongoose.model('NoteBook', NoteBookSchema),
    note: mongoose.model('Note', NoteSchema)
}


